dependencies.yaml

```diff
@@ -0,0 +1,46 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+files:
```

Comment:
For this PR, adding *just enough* in `dependencies.yaml` to set up the environment for conda-based tests in CI.

If we agree on using `rapids-dependency-file-generator` here, I'll do a follow-up PR using this for `pyproject.toml` too.

Reply:
Yes I think so. I think that will become clearer in the follow-up PR I'll add that also manages `pyproject.toml` dependencies... less stuff to keep in sync across multiple places.